client/src/services/data-interface/data-interface.ts

```diff
@@ -241,3 +196,99 @@ export default function getNumOfLicenses(
     return MAX_LICENSES_AVAILABLE - data.length;
   }
 }
+
+interface LicenseAvailability {
+  totalAvailable: number;
+  allAlcoholAvailable: number;
+  beerWineAvailable: number;
+}
+
+export function getAvailableLicensesByZipcode(
+  data: BusinessLicense[],
+  zipcode: EligibleBostonZipcode
+): LicenseAvailability {
+  if (!isEligibleBostonZipCode(zipcode)) {
+    throw new Error(
+      "You must enter a zipcode within the City of Boston. See https://data.boston.gov/dataset/zip-codes/resource/a9b44fec-3a21-42ac-a919-06ec4ac20ab8"
+    );
+  }
+
+  let totalCount = 0;
+  let allAlcCount = 0;
+  let beerWineCount = 0;
+
+  for (const license of data) {
+    if (license.zipcode === zipcode) {
+      totalCount++;
+
+      switch (license.alcohol_type) {
+        case "All Alcoholic Beverages":
+          allAlcCount++;
+          break;
+        case "Wines and Malt Beverages":
+          beerWineCount++;
+          break;
+      }
+    }
+  }
+
+  return {
+    totalAvailable: MAX_AVAILABLE_PER_ZIP - totalCount,
+    allAlcoholAvailable: MAX_ALL_ALC_PER_ZIP - allAlcCount,
+    beerWineAvailable: MAX_BEER_WINE_PER_ZIP - beerWineCount,
+  };
+}
+
+export function getZipcodesWithAvailableLicenses(
+  data: BusinessLicense[],
+  alcoholType?: "All Alcoholic Beverages" | "Wines and Malt Beverages"
+): Array<{
```

Comment:
🌴 🌴 🌴 Non-Blocking comment:
You could leverage the existing `LicenseAvailability` interface if you want. You would gain some flexibility to change types.

```typescript
interface ZipcodeAvailability extends LicenseAvailability {
  zipcode: EligibleBostonZipcode;
}
```